src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,286 @@
+            currTask.markDone();
+
+            returnMessage = "Alright then! I've marked that task as done:" + "\n\t" + currTask;
+            break;
+        case "unmark":
+            index = getIndexFromMessage(message); //get the index
+            currTask = tasks.getTaskAtIndex(index);
+            currTask.markUndone();
+
+            returnMessage = "Alright then! I've marked that task as not done:" + "\n\t" + currTask;
+            break;
+        case "delete":
+            index = getIndexFromMessage(message);
+            currTask = tasks.removeTask(index);
+            storage.modifyStorage(currTask, ConfirmCodes.DELETION, tasks);
+
+            returnMessage = "As you wish. I've removed the task from your list:" + "\n\t" + currTask
+                + "\nI hope it was nothing important..." + "\n" + getTaskCountMessage(tasks);
+            break;
+        case "find":
+            int indexOfSpace = message.indexOf(" ");
+            String keyword = message.substring(message.indexOf(" ") + 1).toLowerCase();
+
+            if (keyword.length() < 1 || indexOfSpace == -1) {
+                throw new DukeException("Pardon me, but the body of the find command should not be empty");
+            }
+
+            String foundTasks = tasks.find(keyword);
+
+            if (foundTasks.equals("")) {
+                returnMessage = "My apologies, but no tasks were found for the given keyword.";
+            } else {
+                returnMessage = "I've searched the archives and found these matching tasks in your list:\n"
+                    + tasks.find(keyword);
+            }
+            break;
+        case "todo":
+            currTask = parseMessageContents(message, TaskTypes.TODO);
+
+            tasks.addTask(currTask);
+            storage.modifyStorage(currTask, ConfirmCodes.ADDITION, tasks);
+
+            returnMessage = "Alright then! I've added the task to your list:" + "\n\t" + currTask
+                + "\n" + getTaskCountMessage(tasks);
+            break;
+        case "deadline":
+            currTask = parseMessageContents(message, TaskTypes.DEADLINE);
+            tasks.addTask(currTask);
+            storage.modifyStorage(currTask, ConfirmCodes.ADDITION, tasks);
+            returnMessage = "Alright then! I've added the task to your list:" + "\n\t" + currTask + "\n"
+                + getTaskCountMessage(tasks);
+            break;
+        case "event":
+            currTask = parseMessageContents(message, TaskTypes.EVENT);
+            tasks.addTask(currTask);
+            storage.modifyStorage(currTask, ConfirmCodes.ADDITION, tasks);
+            returnMessage = "Alright then! I've added the task to your list:" + "\n\t" + currTask + "\n"
+                + getTaskCountMessage(tasks);
+            break;
+        default:
+            throw new DukeException("Pardon me, but I did not understand what you said.");
+        }
+        return returnMessage;
+    }
```

Comment:
If you have time, perhaps you can consider how to use abstraction and encapsulation to simplify repeated code. Long switch statements can be difficult to read at times.

Reply:
Thanks for the good responses. I've noticed you were not allocated to my code, so thank you very much for taking the time to review it 👍